modules/configuring-oidc-authentication.adoc

```diff
@@ -0,0 +1,253 @@
+
+.Prerequisites 
+
+* You have configured Red Hat Single Sign-On for OpenShift. For more information, see link:https://access.redhat.com/documentation/en-us/red_hat_single_sign-on/7.6/html-single/server_installation_and_configuration_guide/index#operator[Red Hat Single Sign-On Operator]. 
+* You have configured TLS/SSL for your {productname} deployment _and_ for Red Hat Single Sign-On. 
+* You have generated a single Certificate Authority (CA) and uploaded it to your Red Hat Single Sign-On Operator _and_ to your {productname} configuration.  
+* You are logged into your {ocp} cluster. 
+* You have installed the OpenShift CLI (`oc`). 
+
+.Procedure 
+
+. Upload a custom SSL/TLS certificate to your {ocp} deployment. 
+
+.. Navigate to the {productname} config tool UI. 
+
+.. Under *Custom SSL Certificates*, click *Select file* and upload your custom SSL/TLS certificates. 
+
+.. Reconfigure your {productname} deployment.
+
+////
+
+.. Base64 encode the content of a single TLS/SSL certificate by entering the following command:
++
+[source,terminal]
+----
+$ cat ssl.crt | base64 -w 0
+----
++
+.Example output
++
+[source,terminal]
+----
+LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSURCVENDQWUyZ0F3SUJBZ0lKQUtsYW9WNHMrb1BtTUEw
+[truncated]
+----
+
+.. Create a Secret called `quay-enterprise-cert-secret` by entering the following command: 
++
+[source,terminal]
+----
+$ oc create create secret generic quay-enterprise-cert-secret --from-file=ca.crt=<path_to_certificate_file>
+----
+
+.. Mount the secret into the `qe` pod in `/conf/stack/extra_ca_certs`:
++
+[source,yaml]
+----
+apiVersion: v1
+data:
+  ssl.crt:  LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSURCVENDQWUyZ0F3SUJBZ0lKQUtsYW9WNHMrb1BtTUE[truncated]
+kind: Secret
+metadata:
+  name: quay-enterprise-cert-secret
+type: Opaque
+----
+
+.. Edit the `quay-enterprise-app` deployment to add the following mount path under `volumeMounts`. This can be done from the {ocp} console or by using `oc edit`.
++
+[source,yaml]
+----
+    - mountPath: /conf/stack/extra_ca_certs
+      name: certvolume
+----
+
+.. Under the `volumes` section, add the `quay-enterprise-cert-secret` as a volume. For example:
++
+[source,yaml]
+----
+    - name: certvolume
+      secret:
+        defaultMode: 420
+        secretName: quay-enterprise-cert-secret
+----
+
+.. Save the manifest. The `Quay` pods should be rescheduled automatically. 
+////
+
+. Log in to the {ocp} web console and navigate to *Operators* -> *Installed Operators* and click on *Red Hat Single Sign-On*. 
+
+. Click on the Red Hat Single Sign-On instance that you will be using for {productname} to access the administration console. 
+
+. Navigate to the Red Hat Single Sign-On *Admin Console*.
+
+.. On the {ocp} *Web Console*, navigate to *Network* -> *Route*. 
+
+.. Select the *Red Hat Single Sign-On* project from the drop-down list. 
+
+.. Find the Red Hat Single Sign-On *Admin Console* in the *Routes* table. 
```

Comment:
Maybe there is something misunderstood about my original comment.  

The current document is viewed like this 
![Screenshot from 2023-08-01 14-48-21](https://github.com/quay/quay-docs/assets/58208478/2a639acf-147f-43a9-87e8-1c9fc3e44f7f)

I think the right way is to delete  the step 1 and the step 2, then the step 3 should be like
```
2.  Follow below steps to find Red Hat Single Sign-On Admin console, the navigate to the Red Hat Single Sign-On Admin Console.

    2.1 On the {ocp} Web Console, navigate to Network → Route.

    2.2 Select the Red Hat Single Sign-On project from the drop-down list.

    2.3 Find the Red Hat Single Sign-On Admin Console in the Routes table.
```

Reply:
BTW,  in final view, there are two `step 1` 

![Uploading Screenshot from 2023-08-01 14-55-27.png…]()

Reply:
oh.  I just notice,  the first `step 1` should not belong to this section.  It should be the `step 2` in section `Configuring the {productname} Operator to use Red Hat Single Sign-On`.  

just like this 

```
Configuring the {productname} Operator to use Red Hat Single Sign-On

Use the following procedure to configure Red Hat Single Sign-On with the {productname} Operator.
Prerequisites

    You have configured the Red Hat Single Sign-On Operator for the {productname} Operator.

Procedure

    1 . Enter the {productname} config editor tool by navigating to Operators → Installed Operators. Click Red Hat Quay → Quay Registry. Then, click the name of your {productname} registry, and the URL listed with Config Editor Endpoint.

    2. Upload the Red Hat Single Sign-On  SSL/TLS  root CA certificate to your {productname}  deployment.

        2.1 Navigate to the {productname} config tool UI.

        2.2 Under Custom SSL Certificates, click Select file and upload your custom SSL/TLS certificates.

        2.3 Reconfigure your {productname} deployment.


    3. Scroll down to the External Authorization (OAuth) section.

    Click Add OIDC Provider.
  ........
```